src/main/java/org/reactome/release/cosmicupdate/COSMICIdentifierUpdater.java

```diff
@@ -0,0 +1,194 @@
+	{
+		// If there is a COSV identifier, we'll update using that.
+		if (this.getCosvIdentifier() != null && !this.getCosvIdentifier().isEmpty())
+		{
+			// Create the InstanceEdit, if necessary.
+			synchronized (COSMICIdentifierUpdater.class)
+			{
+				if (COSMICIdentifierUpdater.instanceEditNewCOSV == null)
+				{
+					COSMICIdentifierUpdater.instanceEditNewCOSV = InstanceEditUtils.createDefaultIE(adaptor, creatorID, true, "Identifier was automatically updated to new \"COSV\" identifier by COSMIC Update process.");
+				}
+			}
+			GKInstance identifierObject = adaptor.fetchInstance( this.getDbID());
+			
+			updateIdentifierObject(adaptor, COSMICIdentifierUpdater.instanceEditNewCOSV, identifierObject, this.getCosvIdentifier());
+		}
+		// If no COSV identifier was found, update using the suggested prefix (determined computationally).
+		else if (this.getSuggestedPrefix() != null && this.getSuggestedPrefix().equalsIgnoreCase(COSMICUpdateUtil.COSMIC_LEGACY_PREFIX))
+		{
+			GKInstance identifierObject = adaptor.fetchInstance( this.getDbID());
+			String currentIdentifier = (String) identifierObject.getAttributeValue(ReactomeJavaConstants.identifier);
+			// If the current identifier already begins with "C" then leave it alone.
+			// This code is for updating numeric identifiers that have a suggested prefix.
+			if (!COSMICUpdateUtil.stringStartsWithC(currentIdentifier.toUpperCase()))
+			{
+				// Create the InstanceEdit, if necessary.
+				synchronized (COSMICIdentifierUpdater.class)
+				{
+					if (COSMICIdentifierUpdater.instanceEditPrependCOSM == null)
+					{
+						COSMICIdentifierUpdater.instanceEditPrependCOSM = InstanceEditUtils.createDefaultIE(adaptor, creatorID, true, "Identifier was automatically prepended with \"COSM\" by COSMIC Update process.");
+					}
+				}
+				updateIdentifierObject(adaptor, COSMICIdentifierUpdater.instanceEditPrependCOSM, identifierObject, this.getSuggestedPrefix() + currentIdentifier);
+			}
+		}
+		// Some identifiers won't have a COSV identifier in the COSMIC files, and they might not have a suggested prefix either.
+		else
+		{
+			logger.info("No suggested prefix OR corresponding COSV identifier for {} (DBID: {}) - identifier will not be updated.", this.getIdentifier(), this.getDbID());
+		}
+	}
+	
+	/**
+	 * Executes an update on an instance.
+	 * Sets the identifier attribute of <code>identifierObject</code> to the value of <code>identifierValue</code>.
+	 * <code>identifierObject</code> (which must be an InstanceEdit) will also have <code>modifiedForCOSMICUpdate</code> added to its <code>modified</code> list.
+	 * The display name of <code>identifierObject</code> will also be regenerated to reflect changes in <code>identifierValue</code>.
+	 * @param adaptor
+	 * @param modifiedForCOSMICUpdate An InstanceEdit which explains why an instance was modified.
+	 * @param identifierObject An object that represents a COSMIC identifier.
```

Comment:
Could the GKInstance class type(s) the identifierObject could be, be listed for context here?

Reply:
There are no hard restrictions on type. It's an object that has a COSMIC identifier.

Reply:
Okay. I think it's most often an EWAS, though.  Maybe an addition to the doc comment - "An object (often an EWAS, but many instance types are possible) that represents a COSMIC identifier.  Personally it helps me orient knowing what type of GKInstance I may or am likely dealing with.

Reply:
I think that the objects here are DatabaseIdentifier objects associated with some other object - _probably_ an EWAS.

Reply:
Okay, so maybe: "An object (most likely a Database Identifier instance) that represents a COSMIC identifier"